src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -1526,6 +1526,78 @@ module Types = struct
 
   let user_command = UserCommand.user_command_interface
 
+  module SnappCommand = struct
+    module Status = struct
+      type t =
+        | Applied
+        | Included_but_failed of Transaction_status.Failure.t
+        | Unknown
+    end
+
+    module With_status = struct
+      type 'a t = { data : 'a; status : Status.t }
+
+      let map t ~f = { t with data = f t.data }
+    end
+
+    let field_no_status ?doc ?deprecated lab ~typ ~args ~resolve =
+      field ?doc ?deprecated lab ~typ ~args ~resolve:(fun c cmd ->
+          resolve c cmd.With_status.data)
+
+    let snapp_command =
+      obj "SnappCommand" ~fields:(fun _ ->
+          [ field_no_status "id" ~typ:(non_null guid) ~args:[]
+              ~resolve:(fun _ parties ->
+                Parties.to_base58_check parties.With_hash.data)
+          ; field_no_status "hash" ~typ:(non_null string) ~args:[]
+              ~resolve:(fun _ parties ->
+                Transaction_hash.to_base58_check parties.With_hash.hash)
+          ; field_no_status "nonce" ~typ:(non_null int) ~args:[]
+              ~doc:
+                "Sequence number of the Snapp transaction for the fee-payer's \
+                 account" ~resolve:(fun _ parties ->
+                Parties.nonce parties.With_hash.data |> Unsigned.UInt32.to_int)
+          ; field_no_status "feePayer" ~typ:(non_null AccountObj.account)
+              ~args:[]
+              ~doc:"Account that pays the fees for the Snapp transaction"
+              ~resolve:(fun { ctx = coda; _ } cmd ->
+                AccountObj.get_best_ledger_account coda
+                  (Parties.fee_payer cmd.With_hash.data))
+          ; field_no_status "accountsAccessed"
+              ~typ:(non_null (list (non_null AccountObj.account)))
+              ~args:[]
+              ~doc:"List of accounts accessed to complete the Snapp transaction"
+              ~resolve:(fun { ctx = coda; _ } parties ->
+                let account_ids =
+                  Parties.accounts_accessed parties.With_hash.data
+                in
+                List.map account_ids ~f:(fun acct_id ->
+                    AccountObj.get_best_ledger_account coda acct_id))
+          ; field_no_status "feeLowerBound" ~typ:uint64 ~args:[]
+              ~doc:
+                "Lower bound on the fee paid by the fee-payer for the Snapp \
```

Comment:
Please include details about why we're setting a lower bound and how someone can interpret what this means for debiting account

Reply:
@mrmr1993 @imeckler Can you suggest wording here?

Reply:
I think we'll probably want to change this so there's an explicit fee. As written, the few payer commits their entire amount regardless of transaction success or failure, and the block producer either
* gets all of it, if the other parties fail
* gets what's left after transfers to other accounts are settled, if the other parties succeed (this amount is this lower bound).

Probably we never want to give a block producer `fee+amount`; once we fix this, we can remove the lower bound and just call this `fee`.

Reply:
For the purposes of this PR, we should leave as is and track it in a separate issue. Does that seem okay @bkase @psteckler?

Reply:
OK with me.

Reply:
Created #9496.